Highlight dates

damlu opened the pull request:
This just adds another parameter `highlightDates` to the `DatePicker` component which accepts an array of Moment objects representing the dates to be highlighted.

Highlighted dates will display with a green background in the calendar.

I've used this in my own log/journal app to indicate days that already have existing entries.  
  

![2016-08-06 14_18_34-localhost_3000](https://cloud.githubusercontent.com/assets/8106053/17459308/188e8882-5be7-11e6-9fb4-8c41fc3aa5d1.png)

martijnrusschen commented:
I like this, this may warrant a new example on the example site (not blocking this PR). Can you please rebase so this is ready to merge?

damlu commented:
Thanks Martijn. I went ahead and updated the docs and examples, then rebased and squashed.

martijnrusschen commented on docs/datepicker.md:
```diff
@@ -53,6 +53,11 @@ type: `func`
 type: `bool`
 
 
+### `highlightDates`
```
This isn't needed. Docs are auto generated with every new release.

damlu replied:
My bad! Reverted that file and updated.

martijnrusschen commented:
Sorry,  I merged again a PR that affects your code. I'm really sorry! If you have no time to update the PR again, I can try doing that before I merge it in. Also, is there a way increase the code coverage of the touched code? It seems that it doesn't hit the target of 75%.

damlu commented:
No worries! I'll have a go at it tomorrow.

damlu commented:
Okay, rebased and fixed the patch coverage. Took the highlightDates parameter out of date_input.jsx as it wasn't actually doing anything there, and so wasn't testable.

martijnrusschen commented:
Awesome!

martijnrusschen commented:
I'll release this soon!
